extensions/arc/src/ui/dashboards/miaa/miaaDashboardOverviewPage.ts

```diff
@@ -40,14 +40,16 @@ export class MiaaDashboardOverviewPage extends DashboardPage {
 	constructor(modelView: azdata.ModelView, private _controllerModel: ControllerModel, private _miaaModel: MiaaModel) {
 		super(modelView);
 		this._instanceProperties.miaaAdmin = this._miaaModel.username || this._instanceProperties.miaaAdmin;
-		this._controllerModel.onRegistrationsUpdated((_: Registration[]) => {
-			this.eventuallyRunOnInitialized(() => {
-				this.handleRegistrationsUpdated().catch(e => console.log(e));
-			});
-		});
-		this._controllerModel.onEndpointsUpdated(endpoints => this.eventuallyRunOnInitialized(() => this.handleEndpointsUpdated(endpoints)));
-		this._miaaModel.onStatusUpdated(status => this.eventuallyRunOnInitialized(() => this.handleMiaaStatusUpdated(status)));
-		this._miaaModel.onDatabasesUpdated(databases => this.eventuallyRunOnInitialized(() => this.handleDatabasesUpdated(databases)));
+		this.disposables.push(
+			this._controllerModel.onRegistrationsUpdated((_: Registration[]) => {
+				this.eventuallyRunOnInitialized(() => {
+					this.handleRegistrationsUpdated().catch(e => console.log(e));
+				});
+			}),
+			this._controllerModel.onEndpointsUpdated(endpoints => this.eventuallyRunOnInitialized(() => this.handleEndpointsUpdated(endpoints))),
+			this._miaaModel.onStatusUpdated(status => this.eventuallyRunOnInitialized(() => this.handleMiaaStatusUpdated(status))),
+			this._miaaModel.onDatabasesUpdated(databases => this.eventuallyRunOnInitialized(() => this.handleDatabasesUpdated(databases)))
+		);
 
 		this.refresh().catch(e => {
```

Comment:
MiaaDashboard already refreshes the miaaModel so I expect these refreshes in the page constructors to go away.

Reply:
Good catch, fixed to only have the dashboard refresh when opened